### ISSUE TITLE
[bug] immutability: false positive on ref.current write inside useEffect

### ISSUE THREAD

**Issue by silverwind** [Rel1cx/eslint-react#1893]
### Describe the bug

Since the rework in https://github.com/Rel1cx/eslint-react/pull/1891 (v5.12.0, still present in v5.13.1), `react-x/immutability` reports writes to `useRef().current` inside a `useEffect` callback. Refs are intentionally mutable outside render; the upstream `react-hooks/immutability` (eslint-plugin-react-hooks 7.1.1) does not report this pattern.

### Reproduction

```tsx
import {useEffect, useRef} from "react";

export function Component() {
  const mounted = useRef<boolean>(false);
  useEffect(() => {
    if (mounted.current) return;
    mounted.current = true;
  }, []);
  return <div/>;
}
```

```
5:13  error  This function may (indirectly) reassign or modify 'mounted' after render, ...  react-x/immutability
7:5   error  This modifies 'mounted'                                                        react-x/immutability
```

### Expected behavior

No errors, matching `react-hooks/immutability` behavior.

### Platform and versions

```
Node v26.5.0
eslint 10.6.0
eslint-plugin-react-x 5.13.1
eslint-plugin-react-hooks 7.1.1 (for comparison)
typescript-eslint 8.63.0
typescript 6.0.3
```

### Additional context

This report was written by Claude on behalf of the reporter.
